snakefmt/formatter.py

```diff
@@ -98,22 +101,23 @@ def format_param(
         result = f"{used_indent}{result}"
         return result
 
-    def format_params(self, parameters: ParameterSyntax) -> str:
-        single_param = False
+    def format_params(self, parameters: ParameterSyntax, in_rule: bool) -> str:
         used_indent = TAB * (parameters.target_indent - 1)
         result = f"{used_indent}{parameters.keyword_name}:{parameters.comment}"
 
-        is_shell = parameters.keyword_name == "shell"
+        p_class = parameters.__class__
+        single_param = issubclass(p_class, SingleParam)
+        # single parameter formatting cancelled if we are in rule-like context and param is not numeric
+        single_param_fting = single_param
```

Comment:
Sorry to be picky could this be `single_param_fmting`? The 'm' adds a lot to the meaning IMO